io-engine/src/bdev/nexus/nexus_child.rs

```diff
@@ -419,6 +431,16 @@ impl<'c> NexusChild<'c> {
         self.sync_state
     }
 
+    /// Return the last fault timestamp of the child.
+    pub fn fault_timestamp(&self) -> Option<DateTime<Utc>> {
+        let ts = self.faulted_at.load(Ordering::Relaxed);
+        if ts == 0 {
+            return None;
+        }
+        NaiveDateTime::from_timestamp_opt(ts as i64, 0)
+            .map(|d| DateTime::from_utc(d, Utc))
+    }
```

Comment:
Nitpicking: you could use more a idiomatic Rust:
`if res == 0 {
    None
} else {
  NaiveDatetime::...
}
`

Reply:
Apparently, I had `else` earlier but IDE suggested me to unwrap the condition and do this way :)